### PR TITLE
Stop participation in the challenge

### PR DESCRIPTION
**Pull request by sanketbansal**
Added feature to stop participation in the challenge by the host.

#### Changes proposed in this pull request:

- Added feature to stop participation in the challenge by the challenge host.

- Link to live demo: http://pr-189-evalai.surge.sh 

Screenshot from 2019-07-25 09-36-12

### CODE REVIEW

File: src/app/components/challenge/challenge.component.ts [hunk 1]
```diff
@@ -286,4 +286,25 @@ export class ChallengeComponent implements OnInit {
...
+      this.apiService.patchUrl(
+        this.endpointsService.editChallengeDetailsURL(this.challenge.creator.id, this.challenge.id),
+        BODY
+      ).subscribe(
+        data => {
+          this.challenge['is_registration_open'] = !this.challenge['is_registration_open'];
+          console.log(data);
+        },
+        err => {
+          console.log(err);
```
Comment (lunayach) [inline src/app/components/challenge/challenge.component.ts]: Remove all the console statements.

File: src/styles/theming.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+@import '~@angular/material/theming';
```
Comment (lunayach) [inline src/styles/theming.scss:1]: Why is this file needed in a "Stop participation" PR?

Comment (sanketbansal) [reply src/styles/theming.scss:1]: This file is added to provide theming for the angular material components of the applications. like checkbox

File: src/app/components/challenge/challenge.component.ts (hunk 1 above)
Comment (sanketbansal) [reply src/app/components/challenge/challenge.component.ts]: done

**Comment by Sanji515** [comment]
Thanks @sanketbansal for the PR, Closing this as it is opened [here](https://github.com/Cloud-CV/EvalAI-ngx/pull/254) after resolving conflicts and with some modifications.
